### PR TITLE
Re-enable client request id tests

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/Azure/autorest.csharp/issues/446

### CODE REVIEW

File: test/AutoRest.TestServer.Tests/AutoRest.TestServer.Tests.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.1.0" />
+    <PackageReference Include="Azure.Core" Version="1.3.0-dev.20200422.1" />
```
Comment (pakrym) [inline test/AutoRest.TestServer.Tests/AutoRest.TestServer.Tests.csproj:11]: Requires latest core

File: test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
+
+            if (ClientRequestIdScope.CurrentClientRequestId != null)
+            {
+                message.Request.ClientRequestId = ClientRequestIdScope.CurrentClientRequestId;
```
Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs:21]: So, why couldn't we do this before? This was what I was thinking originally. Basically, inject a 'replace value' line like this in the pipeline.

Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs:21]: So, why couldn't we do this before? This was what I was thinking originally. Basically, inject a 'replace value' line like this in the pipeline.

File: test/AutoRest.TestServer.Tests/lro.cs [hunk 3]
```diff
@@ -20,45 +20,47 @@ public class LroTest: TestServerTestBase
         public LroTest(TestServerVersion version) : base(version, "lros") { }
 
         [Test]
-        [Ignore("Needs to send x-ms-client-request-id: https://github.com/Azure/autorest.csharp/issues/446")]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/683")]
```
Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/lro.cs:23]: What's the problem with this one?

File: test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs (hunk 2 above)
Comment (pakrym) [reply test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs:21]: We could, with all tests but one that takes it as a parameter

File: test/AutoRest.TestServer.Tests/lro.cs (hunk 3 above)
Comment (pakrym) [reply test/AutoRest.TestServer.Tests/lro.cs:23]: Described in the issue

**Pull request merged**